polar-core/src/kb.rs

```diff
@@ -239,15 +239,17 @@ impl KnowledgeBase {
                     Ok(RuleParamMatch::True)
                 }
             } else {
-                Err(error::OperationalError::InvalidState{msg: format!(
-                    "All registered classes must have a registered MRO. Class {} does not have a registered MRO.",
-                    &rule_instance.tag
-                )}.into())
+                // If `rule_instance.tag` were registered as a class, it would have an MRO.
+                Ok(RuleParamMatch::False(format!(
```

Comment:
Why convert this from an Error?

Reply:
Yeah, I'm just asking whether this should be a failure of the rule to match (`RuleParamMatch::False`) or another type of error indicating that the rule itself is invalid.

If you have registered a constant and then used that as a specializer, isn't that different type of error than just writing a rule that doesn't match a rule type? It's an invalid rule.

Reply:
Talked about this; message is good enough.